### PR TITLE
Infrastructure Encryption

### PR DESCRIPTION
**Pull request by ro100e**
## 🛠️ Description

Added infrastructure encryption parameter to the FinOps Hub template, this will assist customers in removing blockers such as policy requiring infrastructure encryption on storage accounts to successfully deploy hubs.

To deploy yourself:
- [Deploy to Azure](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmicrosoft%2Ffinops-toolkit%2Fb2a744aee1e23c66eca87c18cd4cd0cf68be9ae9%2Fdocs%2Fdeploy%2Ffinops-hub-preview.json/createUIDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fmicrosoft%2Ffinops-toolkit%2Fb2a744aee1e23c66eca87c18cd4cd0cf68be9ae9%2Fdocs%2Fdeploy%2Ffinops-hub-preview.ui.json)

### 🔬 How did you test this change?

> - [ ] 🤏 Lint tests
> - [ ] 🤞 PS -WhatIf / az validate
> - [x] 👍 Manually deployed + verified
> - [ ] 💪 Unit tests
> - [ ] 🙌 Integration tests

### 🙋‍♀️ Do any of the following that apply?

> - [ ] 🚨 This is a breaking change.
> - [x] 🤏 The change is less than 20 lines of code.

### 📑 Did you update `docs/changelog.md`?

> - [x] ✅ Updated changelog (required for `dev` PRs)
> - [ ] ➡️ Will add log in a future PR (feature branch PRs only)
> - [ ] ❎ Log not needed (small/internal change)

### 📖 Did you update documentation?

> - [x] ✅ Public docs in `docs` (required for `dev`)
> - [ ] ✅ Internal dev docs in `src` (required for `dev`)
> - [ ] ➡️ Will add docs in a future PR (feature branch PRs only)
> - [ ] ❎ Docs not needed (small/internal change)

### CODE REVIEW

File: src/templates/finops-hub/modules/hub.bicep [hunk 1]
```diff
@@ -46,6 +46,9 @@ param remoteHubStorageKey string = ''
 @description('Optional. Enable telemetry to track anonymous module usage trends, monitor for bugs, and improve future releases.')
 param enableDefaultTelemetry bool = true
 
+@description('Optional. Enable infrastructure encryption on the storage account.')
+param enableInfrastructureEncryption bool = true
```
Comment (flanakin) [inline src/templates/finops-hub/modules/hub.bicep:50]: Update the default and description here if you end up changing the main.bicep version.

Small nit: Can you move this above the default telemetry parameter? I'd like to keep that last.

```suggestion
 @description('Optional. Enable infrastructure encryption on the storage account. Default = false.')
param enableInfrastructureEncryption bool = false
```

File: src/templates/finops-hub/modules/storage.bicep [hunk 2]
```diff
@@ -36,6 +36,9 @@ param msexportRetentionInDays int = 0
 @description('Optional. Number of months of cost data to retain in the ingestion container. Default: 13.')
 param ingestionRetentionInMonths int = 13
 
+@description('Optional. Enable infrastructure encryption on the storage account.')
+param enableInfrastructureEncryption bool = true
```
Comment (flanakin) [inline src/templates/finops-hub/modules/storage.bicep:40]: Update the default and description here if you end up changing the main.bicep version.

Small nit: Can you move this above the tags parameter? I'd like to keep the storage account configuration settings together.

```suggestion
 @description('Optional. Enable infrastructure encryption on the storage account. Default = false.')
param enableInfrastructureEncryption bool = false
```

File: src/templates/finops-hub/modules/storage.bicep [hunk 3]
```diff
@@ -76,6 +79,10 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' = {
   tags: union(tags, contains(tagsByResource, 'Microsoft.Storage/storageAccounts') ? tagsByResource['Microsoft.Storage/storageAccounts'] : {})
   properties: {
     supportsHttpsTrafficOnly: true
+    encryption: {
+      keySource: 'Microsoft.Storage'
+      requireInfrastructureEncryption: true
```
Comment (flanakin) [inline src/templates/finops-hub/modules/storage.bicep:84]: Should the parameter be used here? Do we need to make the `encryption` property null if this is not enabled?
```suggestion
      requireInfrastructureEncryption: enableInfrastructureEncryption
```

**Review by flanakin — APPROVED** [review]

File: src/templates/finops-hub/main.bicep [hunk 4]
```diff
@@ -45,6 +45,9 @@ param remoteHubStorageUri string = ''
 @secure()
 param remoteHubStorageKey string = ''
 
+@description('Optional. Enable infrastructure encryption on the storage account.')
+param enableInfrastructureEncryption bool = true
```
[the remark this answers is not in the document]
Comment (ro100e) [reply src/templates/finops-hub/main.bicep:49]: lets leave it disabled, and yes tested with PowerBI

File: src/templates/finops-hub/createUiDefinition.json [hunk 5]
```diff
@@ -94,6 +94,26 @@
...
+      {
+        "name": "advanced",
+        "label": "Advanced",
+        "elements": [
+          {
+            "name": "infrastructureEncryptionExplanation",
+            "type": "Microsoft.Common.TextBlock",
+            "visible":true,
+			      "options":{
+				    "text":"Infrastructure encryption can be enabled for the entire storage account. When infrastructure encryption is enabled for a storage account, data is encrypted twice once at the service level and once at the infrastructure level with two different encryption algorithms and two different keys. To enable infrastructure encryption for this storage account, you must check this box at the time that you deploy this template. Infrastructure encryption cannot be enabled or disabled after the account has been created."
```
[the remark this answers is not in the document]
Comment (ro100e) [reply src/templates/finops-hub/createUiDefinition.json]: added documentation for this

**Pull request merged**